backends/p4tools/common/core/target.cpp

```diff
@@ -48,23 +47,95 @@ bool Target::init(std::string deviceName, std::string archName) {
     return false;
 }
 
-bool Target::setDevice(std::string deviceName) {
-    std::transform(deviceName.begin(), deviceName.end(), deviceName.begin(), ::tolower);
-    if (defaultArchByDevice.count(deviceName) == 0U) {
+std::optional<ICompileContext *> Target::initializeTarget(std::string_view toolName,
+                                                          std::string_view target,
+                                                          std::string_view arch) {
+    // Establish a dummy compilation context so that we can use ::error to report errors while
+    // processing target and arch.
+    class DummyCompileContext : public BaseCompileContext {
+    } dummyContext;
```

Comment:
Any reason to not instantiate `BaseCompileContext` directly?

Reply:
The constructor is protected.